### PR TITLE
Fixes #3652. Setting Menus causes unexpected Exception.

### LINKED ISSUE #3644: Some uses of the `CheckStyle` enum are buggy
Via @bdisp:      @tig after the creation of the `CheckStyle` enum I already saw some bug on getting the right bool value for bool properties, like the following:

```cs
IsTopLevel = _frmMenuDetails.CkbIsTopLevel?.State == CheckState.UnChecked,
HasSubMenu = _frmMenuDetails.CkbSubMenu?.State == CheckState.UnChecked,

```
As you can see if any of the properties are `false` they will be set incorrectly as `true` because the `State` will always be equal to `CheckState.UnChecked`.

_Originally posted by @BDisp in https://github.com/gui-cs/Terminal.Gui/issues/3634#issuecomment-2267959850_

### LINKED ISSUE #3652: Setting `Menus` causes unexpected Exception
**Describe the bug**

MenuBar.Menus magic property behaves inconsistently when set multiple times.

The setter should replace existing values with the new values. 

Regarding the new Exception that is throw, it is problematic for designer where user might easily create multiple menu items with the same name or shortcut key.  

For this use case we need a setting which suppress the Exception.

Test
```csharp
    [Fact]
    public void Test_SetMenus()
    {
        var mb = new MenuBar ();

        var i1 = new MenuBarItem ("heey", "fff", () => { }, () => { return true; })
        {
            HotKey = new Rune('I')
        };

        mb.Menus = new MenuBarItem [] { i1 };
        mb.Menus = new MenuBarItem [] { i1 };
    }
```
Throws
```csharp
  Message: 
System.InvalidOperationException : A key binding for i exists (KeyBinding { Commands = Terminal.Gui.Command[], Scope = Focused, BoundView = , Context = 0 }).

  Stack Trace: 
KeyBindings.Add(Key key, KeyBinding binding, View boundViewForAppScope) line 32
MenuBar.set_Menus(MenuBarItem[] value) line 178
MenuTests.Test_SetMenus() line 29
RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
```

### PR DESCRIPTION
**Pull request by BDisp**
## Fixes

- Fixes #3652, Fixes #3644

## Proposed Changes/Todos

- [x] Rename Shortcut to ShortcutKey and change type to Key
- [x] Change HotKey type to Key
- [x] Improving add and remove menu items dynamically
- [x] Allow run an action in IsTopLevel menu bar item through HotKey

## Pull Request checklist:

- [x] I've named my PR in the form of "Fixes #issue. Terse description."
- [x] My code follows the [style guidelines of Terminal.Gui](https://github.com/gui-cs/Terminal.Gui/blob/develop/.editorconfig) - if you use Visual Studio, hit `CTRL-K-D` to automatically reformat your files before committing.
- [x] My code follows the [Terminal.Gui library design guidelines](https://github.com/gui-cs/Terminal.Gui/blob/develop/CONTRIBUTING.md)
- [x] I ran `dotnet test` before commit
- [x] I have made corresponding changes to the API documentation (using `///` style comments)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any poor grammar or misspellings
- [x] I conducted basic QA to assure all features are working

### CODE REVIEW

File: Terminal.Gui/Input/Key.cs [hunk 1]
```diff
@@ -971,4 +971,19 @@ out parsedInt
     public static Key F24 => new (KeyCode.F24);
 
     #endregion
+
+    private static Rune _shortcutDelimiter = new ('+');
+
+    /// <summary>Sets or gets the shortcut delimiter separator. The default is "+".</summary>
+    public static Rune ShortcutDelimiter
```
Comment (tig) [inline Terminal.Gui/Input/Key.cs:978]: We should rename this to "Separator". With this PR, this is now used for all `Key` usages and should be more generic.

We should also make this a CM property. It needs to be configurable.

On Windows the standard is +.

On Mac the standard is -.

**Review by tig — CHANGES_REQUESTED** [review]
Really great work!

File: Terminal.Gui/Input/Key.cs (hunk 1 above)
Comment (BDisp) [reply Terminal.Gui/Input/Key.cs]: > We should rename this to "Separator". With this PR, this is now used for all `Key` usages and should be more generic.
> 
Done.

> We should also make this a CM property. It needs to be configurable.
> 
Done.

> On Windows the standard is +.
> 
Done.

> On Mac the standard is -.

I think this is irrelevant and I have no idea how to configure this on the `config.json` to achieve this.

File: Terminal.Gui/Input/Key.cs [hunk 2]
```diff
@@ -972,17 +973,18 @@ out parsedInt
 
     #endregion
 
-    private static Rune _shortcutDelimiter = new ('+');
+    private static Rune _separator = new ('+');
 
     /// <summary>Sets or gets the shortcut delimiter separator. The default is "+".</summary>
```
Comment (tig) [inline Terminal.Gui/Input/Key.cs:978]: "Gets or sets the separator character used when parsing and printing Keys. E.g. <c>Ctrl+A</c>. The default is '+'".

File: Terminal.Gui/Input/Key.cs [hunk 3]
```diff
@@ -450,7 +451,7 @@ public override bool Equals (object obj)
 
     /// <summary>Pretty prints the KeyEvent</summary>
```
Comment (tig) [inline Terminal.Gui/Input/Key.cs:452]: KeyEvent->Key

File: Terminal.Gui/Input/Key.cs [hunk 4]
```diff
@@ -972,17 +973,18 @@ out parsedInt
...
     {
-        get => _shortcutDelimiter;
+        get => _separator;
         set
         {
-            if (_shortcutDelimiter != value)
+            if (_separator != value)
             {
-                _shortcutDelimiter = value == default (Rune) ? new ('+') : value;
+                _separator = value == default (Rune) ? new ('+') : value;
```
Comment (tig) [inline Terminal.Gui/Input/Key.cs:987]: I don't think this test for default is needed. _separator is set to "+" above.

**Review by tig — CHANGES_REQUESTED** [review]
Some minor tweaks please.

File: Terminal.Gui/Input/Key.cs (hunk 4 above)
Comment (BDisp) [reply Terminal.Gui/Input/Key.cs:987]: I added a unit test that proves it's needed to test for `Rune` default, otherwise `Ctrl+A` will only print `Ctrl`.

**Pull request merged**